### DISCUSSION TITLE
Central broadcasting studio, listeners should use relay in the cloud. Use AzuraRelay for this setup?

### DISCUSSION THREAD

**Discussion by Moonbase59** [AzuraCast/AzuraCast#6801] [Q&A]
I have the following setup and would like to know if [AzuraRelay](https://github.com/AzuraCast/AzuraRelay) would be good for this, or if I should setup my own relaying?

Central broadcasting location
- Central AzuraCast
  - 3 stations
  - 13 streams total (2 video (MPEG-TS over Icecast), 3 HLS, misc. Icecast MP3, AAC)
  - ~160.000 songs (central music database on NAS, NFS-mounted as read-only, used by all stations)
  - ~60 playlists per station
- Broadcasting studio, feeds DJ input on AzuraCast and studio video feed
- Linux DBUS, MQTT, Home Assistant, StudioDisplay and KODI integration, takes care of
  - switching off studio monitors when mic(s) open
  - pausing KODI when mic(s) open (in case KODI is used to follow news/sports broadcasts)
  - phone request line integration
  - signalling tower, On-Air lights (both door and "virtual" software signalling), incoming phone call signal, studio ready, streaming status
  - StudioDisplay monitor (studio clock, weather, now playing on station & KODI, Studio Ready, Streaming, Microphone and Request line indicators)
- All content (audio, video, web site updates) generated here.
- External DJs connect here to stream/send over their shows & studio cam streams

Since this is the most appropriate setup to have everything local, and fast access to terabytes of data, I wonder if the following could be made using AzuraRelay:
1. External DJs still connect to the main AzuraCast (here), both audio & video, and have their broadcasts recorded in the local AzuraCast instance.
1. This AzuraCast only provides "master streams" for the relay (both video & audio), listeners should _not_ connect to this instance.
1. An external (Digital Ocean) (AzuraRelay?) relay should handle _all_ listener streams, both video and audio. A listener should _only_ be able to connect to the external relay.
1. The external relay should be able to relay video (both MPEG-TS via Icecast and HLS) and audio (HLS and Icecast MP3/AAC streams).
1. All public player & website links should _only_ show and allow stream selection from the external relay.
1. Using more than one relay should be possible.
1. Aggregate listener statistics (Icecast format is good enough) from all relays plus local AzuraCast must be available at one single point. Ideally, this should also provide a "well-known" format (like Icecast stats XML) so DJ’s/station owners can see it in their broadcasting software (like IDJC, mAirList, SAM).
1. More detailed reports (SoundExchange, GEMA/GVL, …) would be a nice extra.

This setup should allow central & fast access to everything, while offloading the streaming load to an external relay. Best of both worlds, I should think. (My Internet upload would allow only ~250 listeners, no video included, so all this must be done on a high-bandwidth external relay.)

**Would that be a setup AzuraRelay can provide?**

@BusterNeece:
- To what extent does an AzuraCast + AzuraRelay combo already provide this?
- Do you see any point that _won’t ever_ be possible using AC/AR?

**Comment by BusterNeece** [comment]
@Moonbase59 With regard to the one requirement:

> The external relay should be able to relay video (both MPEG-TS via Icecast and HLS) and audio (HLS and Icecast MP3/AAC streams).

AzuraRelay can't currently relay video as far as I know (not without custom configuration), and it also doesn't currently originate HLS as that would be a much more difficult thing to mirror from the parent server compared to the Icecast stream.

**Reply by amarmashruwala** [reply]
How about if you used mediamtx for your video streams?
